lib/reporters/base.js

```diff
@@ -225,11 +226,21 @@ exports.list = function(failures){
  * @api public
  */
 
-function Base(runner) {
+function Base(runner, options, path) {
   var self = this
     , stats = this.stats = { suites: 0, tests: 0, passes: 0, pending: 0, failures: 0 }
     , failures = this.failures = [];
 
+  this.options = options;
+  this.path = path;
+
+  if (path) {
+    if (!fs.createWriteStream) {
```

Comment:
Yeah, the idea of a generic writable stream hit me too ... but I'm hesitant to do much refactoring though, given how long this pull request has been floating around (almost a year). If there's a way to get it out as-is, and follow-up with improvements, I'm more keen to that.